fw/sock_clnt.c

```diff
@@ -88,10 +88,15 @@ tfw_cli_conn_alloc(int type)
 		return NULL;
 
 	tfw_connection_init((TfwConn *)cli_conn);
+	cli_conn->ws_lock = kzalloc(sizeof(tfw_ws_spinlock_t), GFP_ATOMIC);
```

Comment:
We need to check for allocation failure here.

Reply:
Yeah. Still, this whole lock/refcnt dancing business looks fishy at first glance. I am not convinced it does reliably what it's intended to do. I didn't analyze it deeply though so there's always a chance I am wrong.

Reply:
The main problem is that we operate with a paired connection in the drop of another and a race can occur.

Reply:
Hi, @keshonok!
It seems to have been reasonable. look diagonally please